src/gtfs/read.rs

```diff
@@ -576,6 +576,20 @@ where
 {
     let filename = "agency.txt";
     let gtfs_agencies = read_objects::<_, Agency>(file_handler, filename)?;
+
+    for (current_agency, next_agency) in gtfs_agencies.iter().zip(gtfs_agencies.iter().skip(1)) {
```

Comment:
Why not just compare all others to the first one (that would be the only referent, and it would save one iterator) ?
It looks weird to me to use a different "referent" each time we compare (in the current version), but it works.

Do we want to point out the less-represented timezone ? Not sure it's super useful, and it would require more code.

Reply:
@pbougue 
It breaks at the first difference, and without memory allocation :thinking:

Reply:
To clarify my comment after discussion, I suggest comparing only `gtfs_agencies.first()` with all the following.
Something along that (not tested at all, maybe store only referent timezone, not agency):
```rust
let ref_agency = gtfs_agencies.first();

for agency in gtfs_agencies.iter().skip(1) {
    if ref_agency.timezone != agency.timezone {
        [...]
    }
}
```

Anyway it's just a suggestion not a veto as current code will warn if and only if problems arise.

Reply:
the number of iterations will be the same.
If we have a, b, ,c and d

my proposal will give : 
 a compare with b
 b compare with c
 c compare with d

your proposal will give : 
 a compare with b
 a compare with c
 a compare with d

Reply:
I like @pbougue's solution because:
- it's one iterator instead of 2 iterators
- it feels easier to understand what we're trying to do (the `.zip()` feel a little less natural)

But in any case, I'm fine with whatever we end up merging.

Reply:
I capitulate because it is more readable